Assets/Scripts/Entity/Player/Sanity/SanityController.cs

```diff
@@ -16,64 +14,270 @@ public enum StatType { Health, Saturation, Hydration };
+            AttackLogic.OnPlayerAttackPerformed += OnAttackPerformed;
+            PlayerState.OnPlayerHit += OnPlayerHit;
+            PlayerState.OnPlayerHealed += OnPlayerHealed;
         }
 
         private void OnDisable()
         {
             PlayerState.OnPlayerHealthUpdate -= (int health) => sanityMath.InfluenceSanityByStat(StatType.Health, health);
             PlayerState.OnPlayerSaturationUpdate -= (int saturation) => sanityMath.InfluenceSanityByStat(StatType.Saturation, saturation);
             PlayerState.OnPlayerHydrationUpdate -= (int hydration) => sanityMath.InfluenceSanityByStat(StatType.Hydration, hydration);
+            AttackLogic.OnPlayerAttackPerformed -= OnAttackPerformed;
+            PlayerState.OnPlayerHit -= OnPlayerHit;
+            PlayerState.OnPlayerHealed -= OnPlayerHealed;
         }
 
         /// <summary>
-        /// Tick sanity.
+        /// Tick sanity. Influenced by character needs (health, saturation, hydration)
+        /// and character events (like hit, fight, heal, etc.).
         /// </summary>
         private void Update()
         {
+            // character needs
             TickSanity();
-        }
 
-        
+            // character events
+            if (_isFighting)
+            {
+                _fightTime += Time.deltaTime;
+                if (_fightTime >= criticalFightTime)
+                {
+                    // check if last attacked target was aggressive
+                    if (_enemy.IsAggressive)
+                        AddUpEventTick(fightMalus);
+                    else
+                        AddUpEventTick(fightNonAggressivesMalus);
+                    // start new fight
+                    _fightTime = 0;
+                }
+                MightEndFighting();
+            }
+
+            if (_healBonus)
+                MightEndHealBonus();
+
+            if (applyCampfireBonus)
+                CheckCampfireRange();
+        }
 
         /// <summary>
-        /// Sums up a tick weighting the total change rate with <see cref="severity"/>. 
-        /// Only sums up a positive tick if player doesn't have full sanity. Ticks are either
-        /// positive or negative 1. If summed up tick, changes player sanity.
+        /// Depending on the player's needs (health, saturation, hydration), sum up a tick 
+        /// weighting the total change rate with <see cref="severity"/>. 
+        /// Only sum up a positive tick if player doesn't have full sanity. Ticks are either
+        /// positive or negative 1. If summed up tick, change player sanity.
         /// </summary>
         private void TickSanity()
         {
-            _nextTick += sanityMath.GetCurrentChange();
-            if (_playerState.Sanity == 100 && _nextTick > 0)
+            _needsTick += sanityMath.GetCurrentChange();
+            if (_playerState.Sanity == 100 && _needsTick > 0)
             {
                 // don't build up a positive tick value when sanity == 100
```

Comment:
why no build up at max sanity? :)

Reply:
wow that's some old stuff xD because it'd be kinda broken if your sanity ticks from 100 to 99 and instantly back to 100 because you stacked up some boni